pymeasure/instruments/hp/hp4145x.py

```diff
@@ -0,0 +1,1252 @@
+        smu.save(['VC', 'IC', 'VB', 'IB'])
+
+        # take measurements
+        status = smu.measure()
+
+        # measured data is a pandas dataframe and can be exported to csv.
+        data = smu.get_data(path='./t1.csv')
+
+    The JSON file is an ascii text configuration file that defines the settings of each channel
+    on the instrument. The JSON file is used to configure the instrument using the convenience
+    function :meth:`~.HP4145x.configure` as shown in the example above. For example, the
+    instrument setup for a bipolar transistor measurement is shown below.
+
+    .. code-block:: json
+
+         {
+                "SMU1": {
+                    "voltage_name" : "VC",
+                    "current_name" : "IC",
+                    "channel_function" : "VAR1",
+                    "channel_mode" : "V",
+                    "series_resistance" : "0OHM"
+                },
+
+                "SMU2": {
+                    "voltage_name" : "VB",
+                    "current_name" : "IB",
+                    "channel_function" : "VAR2",
+                    "channel_mode" : "I",
+                    "series_resistance" : "0OHM"
+                },
+
+                "SMU3": {
+                    "voltage_name" : "VE",
+                    "current_name" : "IE",
+                    "channel_function" : "CONST",
+                    "channel_mode" : "V",
+                    "constant_value" : 0,
+                    "compliance" : 0.1
+                },
+
+                 "SMU4": {
+                    "voltage_name" : "VS",
+                    "current_name" : "IS",
+                    "channel_function" : "CONST",
+                    "channel_mode" : "V",
+                    "constant_value" : 0,
+                    "compliance" : 0.1
+                },
+
+                "VAR1": {
+                    "start" : 1,
+                    "stop" : 2,
+                    "step" : 0.1,
+                    "spacing" : "LINEAR",
+                    "compliance" : 0.1
+                },
+
+                "VAR2": {
+                    "start" : 0,
+                    "step" : 10e-6,
+                    "points" : 3,
+                    "compliance" : 2
+
+                }
+            }
+
+    """
+
+    def __init__(self, adapter, name="HP 4145A/4145B Semiconductor Parameter Analyzer",
+                 **kwargs):
+        kwargs.setdefault('timeout', 100000)
+        super().__init__(
+            adapter,
+            name,
+            **kwargs
+        )
+
+        self._mode = None
+        self.smu1 = SMU(self, 1)
+        self.smu2 = SMU(self, 2)
+        self.smu3 = SMU(self, 3)
+        self.smu4 = SMU(self, 4)
+        self.vmu1 = VMU(self, 1)
+        self.vmu2 = VMU(self, 2)
+        self.vsu1 = VSU(self, 1)
+        self.vsu2 = VSU(self, 2)
+        self.var1 = VAR1(self)
+        self.var2 = VAR2(self)
+        self.vard = VARD(self)
```

Comment:
See here how to add channels to an instrument: https://pymeasure.readthedocs.io/en/latest/dev/adding_instruments/channels.html

That way, the channels are there already in the class (not only instance), and the channels are in a dictionary, which will make group actions (like the `manual_flush` property) easier.